lib/oneview-sdk/rest.rb

```diff
@@ -121,6 +123,75 @@ def rest_delete(path, options = {}, api_ver = @api_version)
       rest_api(:delete, path, options, api_ver)
     end
 
+    # Uploads a file to a specific uri
+    # @param [String] file_path
+    # @param [String] uri The uri starting with "/"
+    # @param [Hash] body_params The params to append to body of http request. Default is {}.
+    # @option body_params [String] 'name' The name to show (when resource accepts a name)
+    # @param [Integer] timeout The number of seconds to wait for completing the request. Default is 300.
+    # @return [OneviewSDK::Resource] if the upload was sucessful, return a Resource object
+    def upload_file(file_path, uri, body_params = {}, timeout = READ_TIMEOUT)
+      raise NotFound, "ERROR: File '#{file_path}' not found!" unless File.file?(file_path)
+      options = {
+        'Content-Type' => 'multipart/form-data',
+        'X-Api-Version' => @api_version.to_s,
+        'auth' => @token
+      }
+      url = URI.parse(URI.escape("#{@url}#{uri}"))
+
+      File.open(file_path) do |file|
+        name_to_show = body_params.delete('name') || body_params.delete(:name)
+        name_to_show = name_to_show ? name_to_show + File.extname('/tmp/artifact_bundle.zip') : File.basename(file_path)
+        body_params['file'] = UploadIO.new(file, 'application/octet-stream', name_to_show)
+        req = Net::HTTP::Post::Multipart.new(
+          url.path,
+          body_params,
+          options
+        )
+
+        http_request = Net::HTTP.new(url.host, url.port)
+        http_request.use_ssl = true
```

Comment:
Verify [here](https://github.com/HewlettPackard/oneview-sdk-ruby/blob/master/lib/oneview-sdk/image-streamer/resource/api300/golden_image.rb#L109). It would be nice to check if url.scheme is `https` and do the same thing that was done in GoldenImage.